From f131eccfafa183ae126c475283a6104c5f0c750d Mon Sep 17 00:00:00 2001
From: Rachapalli Bandi Jagadeesh <jagadeesh.rachapalli.bandi@intel.com>
Date: Fri, 12 Jun 2026 18:10:00 +0530
Subject: [PATCH] Don't use match_token_uuid for 6.6 kernel
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

struct vfio_device_ops has no member named ‘match_token_uuid’ for 6.6 kernel.
Hence observing below errors.
To fix these errors, don't use match_token_uuid and related function for 6.6
kernel. On kernels < 6.2, the full VFIO framework is backported
(CPTCFG_BUILD_FULL_VFIO) which includes match_token_uuid in vfio_device_ops.

Errors:
--------------------------------------------------------------------------------
error: ‘const struct vfio_device_ops’ has no member named ‘match_token_uuid’
error: initialization of ‘void (*)(struct vfio_device *, u64,  u64)’ {aka ‘
void (*)(struct vfio_device *, long long unsigned int,  long long unsigned int
)’} from incompatible pointer type ‘int (*)(struct vfio_device *, const uuid_t *)
--------------------------------------------------------------------------------

Reference:
	86624ba3b522 vfio/pci: Do vf_token checks for VFIO_DEVICE_BIND_IOMMUFD

Reviewed-by: S A Muqthyar Ahmed <syed.abdul.muqthyar.ahmed@intel.com>
coauthered-by: Rachapalli Bandi Jagadeesh <jagadeesh.rachapalli.bandi@intel.com>
Signed-off-by: Kanaka Raju Nayana <kanaka.raju.nayana@intel.com>
---
 drivers/vfio/pci/xe/main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/vfio/pci/xe/main.c b/drivers/vfio/pci/xe/main.c
index 606b4e2..a7d7bb9 100644
--- a/drivers/vfio/pci/xe/main.c
+++ b/drivers/vfio/pci/xe/main.c
@@ -505,7 +505,9 @@ static const struct vfio_device_ops xe_vfio_pci_ops = {
 	.mmap = vfio_pci_core_mmap,
 	.request = vfio_pci_core_request,
 	.match = vfio_pci_core_match,
+#if !defined(BPM_MATCH_TOKEN_UUID_NOT_PRESENT) || defined(CPTCFG_BUILD_FULL_VFIO)
 	.match_token_uuid = vfio_pci_core_match_token_uuid,
+#endif
 	.bind_iommufd = vfio_iommufd_physical_bind,
 	.unbind_iommufd = vfio_iommufd_physical_unbind,
 	.attach_ioas = vfio_iommufd_physical_attach_ioas,
-- 
2.43.0

